cpp/src/unary/math_ops.cu

```diff
@@ -295,7 +295,11 @@ std::unique_ptr<column> unary_op_with(column_view const& input,
     input.type(), input.size(), copy_bitmask(input, stream, mr), input.null_count(), stream, mr);
 
   auto out_view = result->mutable_view();
-  Type const n  = std::pow(10, -input.type().scale());
+
+  Type n = 10;
+  for (int i = 1; i < -input.type().scale(); ++i) {
+    n *= 10;
```

Comment:
do we need to worry about overflow?

Reply:
No, I don't think so. We know that the value `10^-scale` is representable by the device storage type (e.g. `__int128_t` or `int64_t`). And in the worst case, this won't introduce a _new_ overflow if one already existed.